salt/states/esxcluster.py

```diff
@@ -0,0 +1,522 @@
+# -*- coding: utf-8 -*-
+'''
+Manage VMware ESXi Clusters.
+
+Dependencies
+============
+
+- pyVmomi Python Module
+
+
+pyVmomi
+-------
+
+PyVmomi can be installed via pip:
+
+.. code-block:: bash
+
+    pip install pyVmomi
+
+.. note::
+
+    Version 6.0 of pyVmomi has some problems with SSL error handling on certain
+    versions of Python. If using version 6.0 of pyVmomi, Python 2.6,
+    Python 2.7.9, or newer must be present. This is due to an upstream dependency
+    in pyVmomi 6.0 that is not supported in Python versions 2.7 to 2.7.8. If the
+    version of Python is not in the supported range, you will need to install an
+    earlier version of pyVmomi. See `Issue #29537`_ for more information.
```

Comment:
Fair enough but are we detecting this condition and warning the user at all? It's a bit much to ask them to read the docs here to determine what could be a very hard-to-detect condition. I wonder if we could improve this at all.

Reply:
I am not sure how to do this though. The only thing I can think of is warnings in tests if there is an incompatibility, but there is no guarantee that everyone will run the tests.

Reply:
Can we detect this in `__virtual__()` and `return False, 'State module did not load: Incompatible versions of Python and pyVmomi present'`, or something along those lines?

Also, this is minor, but I think the mention of Python 2.6 can be removed to reduce potential confusion. Support for Python 2.6 in salt has been removed as of the 2017.7.0 release.

Reply:
We should be able to. This needs to happen in all states though and the `vsphere` execution module. I will add it in this one to `esxcluster` and I will create a separate PR to fix the rest

Reply:
Ok, I have a change - not as trivial as expected. Testing it out now. Question is if we should add it in the state files or just in the `vsphere` execution module where pyVmomi is actually used. @rallytime, @cachedout what do you think?

Reply:
Ok did both. Let's see if the tests pass. From what I remember they don't actually test the `__virtual__` function